### DISCUSSION TITLE
What's the difference between the gossipsub api in go and rust?

### DISCUSSION THREAD

**Discussion by harsh4786** [libp2p/rust-libp2p#4442] [Q&A] [answered]
Specifically, does rust gossipsub support `Join()` like as seen in go libp2p? Or ,
`subscribe()` does exactly that behind the scenes?

**Comment by thomaseizinger** [comment]
As far as I understand, JOIN is an implementation detail of gossipsub and refers to a network message but it is not exposed explicitely.

See https://github.com/libp2p/specs/blob/master/pubsub/gossipsub/gossipsub-v1.0.md#topic-membership.

**Reply by harsh4786** [reply]
so does `subscribe()` internally call `Join()`?

**Reply by kehiy** [reply, marked as answer]
hello @harsh4786 !

The `subscribe()` in rust implementation does the same work as `Join()` in go.
Both are used to indicate an interest in a particular topic and receive messages related to it. Therefore, it is safe to say that `subscribe()` in Rust performs the same action as `Join()` in Go.
